server/src/rpc.rs

```diff
@@ -533,9 +536,14 @@ fn sort_custom_types(
         graph.add_node(ty.name.as_str());
         ty_pos.insert(ty.name.as_str(), pos);
         for field in &ty.field_defs {
-            if ts.lookup_builtin_type(&field.field_type).is_err() {
-                graph.add_node(field.field_type.as_str());
-                graph.add_edge(field.field_type.as_str(), ty.name.as_str(), ());
+            let field_type = field.field_type()?;
+            if !field_type.is_builtin(ts) {
+                let entity_name = match field_type {
+                    TypeEnum::Entity(name) => name,
+                    _ => anyhow::bail!("if type is not a builtin, it must be an entity"),
+                };
+                graph.add_node(entity_name);
+                graph.add_edge(entity_name, ty.name.as_str(), ());
```

Comment:
Actually, the error branch should be unreachable, I think it's cleaner this way:
```suggestion
match field_type {
    TypeEnum::Entity(name) if field_type.is_builtin(ts) => {
                graph.add_node(entity_name);
                graph.add_edge(entity_name, ty.name.as_str(), ());
    }
    _ => ()
```